### ISSUE TITLE
Global: Unicode filename support + Obsidian Japanese Mac detection

### ISSUE THREAD

**Issue by rparnas** [elasota/scummvm#150]
At a minimum:
- Add a detection table entry, which following the pattern of other games, will have a puny-encoded unicode filename for the Obsidian Installer. (`Common::punycode_decodefilename` can decode those).
- Update `void ObsidianGameDataHandler::unpackMacRetailInstaller(Common::Array<Common::SharedPtr<ProjectPersistentResource>> &persistentResources, Common::Array<FileIdentification> &files)` to check for the new installer filename.
- Update `void ObsidianGameDataHandler::categorizeSpecialFiles(Common::Array<FileIdentification> &files)` to check for the new filename.

Since unicode game filenames seem rare it doesn't seem like there are solid patterns for them in the codebase. Maybe FileIdentification.fileName should be changed to be a Common::U32String, I have no idea. If that's the case maybe Common::Path needs a new implicit cast or something (or is the file IO just supposed to detect punycode encoding... I'm lost with all these string types).

Also it was suggested to use the pattern `str.encode().c_str()` to pass as an argument to `error` but though it outputs correct UTF-8 to the log, the debug console seems to choke.

Regardless, I hacked up boot.cpp to successfully completed the Mac Japanese version with no issues. The hack is probably not useful but it's [here](https://github.com/rparnas/scummvm/commit/9a29f37f3525b785df12c7a11008a41dc392f567) for reference.

```
	{ // Obsidian, Japanese Macintosh, dumped
		{
			"obsidian",
			"CD",
			{
				{ "xn--u9j9ecg0a2fsa1io6k6jkdc2k", MTFT_SPECIAL, "af62516a9a9bd16bc5c01e755a9f7329", 9186432 }, // オブシディアンのインストール
				{ "Obsidian Data 2", MTFT_ADDITIONAL, "d3b4746dd05adba87e15f83a1599c6fe", 570855424 },
				{ "Obsidian Data 3", MTFT_ADDITIONAL, "96bb5f6ab893aca9e8e3ce19d19974fc", 616954880 },
				{ "Obsidian Data 4", MTFT_ADDITIONAL, "54cf1745ea19e0cc33d76905d0b9fc41", 593788928 },
				{ "Obsidian Data 5", MTFT_ADDITIONAL, "adc938b08ba2d2f0a08c934e6105b0f9", 583581824 },
				{ "Obsidian Data 6", MTFT_ADDITIONAL, "09de40ab9001eda30291421d2736fa76", 551191680 },
				AD_LISTEND
			},
			Common::JA_JPN,
			Common::kPlatformMacintosh,
			ADGF_TESTING,
			GUIO3(GAMEOPTION_WIDESCREEN_MOD, GAMEOPTION_AUTO_SAVE_AT_CHECKPOINTS, GAMEOPTION_SOUND_EFFECT_SUBTITLES)
		},
		GID_OBSIDIAN,
		0,
		0,
	},
```

**Comment by elasota** [comment]
I think this is supposed to be supported if the filenames are UTF-8?

**Comment by rparnas** [comment]
Non-C++ programmer here: I don't know, I got lost trying to implement this because it seems to require architectural decisions about string/path handling.

I suggest renaming the English installer to the name of the Japanese installer, "オブシディアンのインストール", and making a temporary detection entry for that, so you can see the issues yourself. (Some of the other engines' detection tables seem to use puny-encoded strings for that, but I have no prejudice towards if that is the best way to do it).

Also, console fonts don't have all unicode characters, so an error regarding this filename will be gibberish, but the filename should be correctly written to the log.
